src/crimson/mon/MonClient.cc

```diff
@@ -76,16 +96,25 @@ class Connection {
 
 private:
   bool closed = false;
+  // v1
   seastar::promise<Ref<MAuthReply>> reply;
+  // v2
+  utime_t auth_start;
+  ceph::auth::method_t auth_method = 0;
+  seastar::promise<> auth_done;
+  AuthRegistry* auth_registry;
```

Comment:
Do we have any business in storing nullified `auth_registry`? if not, then maybe a reference could do instead?

In a latter commit also v1 starts relying on `auth_registry` via the `setup_session()` method, so the member becomes shared between v1 and v2 IIUC.

Reply:
yeah. i added a commit to address this.